src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1536,10 +1537,10 @@ ConvToMatmul::ConvToMatmul(Context::Ref ctx) {
             auto new_reshape2 = std::make_shared<ov::op::v1::Reshape>(matched_node_param2, new_const2, false);
 
             // Connect to Reshape
-            if (ov::op::util::is_parameter(matched_node_param2)) {
+            if (matched_node_param2_out == matched_node_multiply) {
                 matched_node_multiply->input(1).replace_source_output(new_reshape2);
                 matched_node_multiply->validate_and_infer_types();
-            } else {  // constant -> convert -> multiply
+            } else {
```

Comment:
But some comments documenting what cases you're looking at still would help a lot

Reply:
Done